### ISSUE TITLE
Revival Protocol cures the same statuses that Medikit does, if able

### ISSUE THREAD

**Issue by copyrite** [X2CommunityCore/X2WOTCCommunityHighlander#1435]
According to localization, Revival Protocol only removes mental impairments. What it doesn't tell you is that it also cures those statuses that a Medikit would. However, being afflicted by those statuses doesn't make a unit an eligible target for Revival Protocol. Therefore, there is a three-way conflict between the localization, targeting conditions, and removed effects.

Suggested fix is to align with localization, since that's the best guess of Firaxis' intent.

Related, if not directly applicable code from LWOTC here: https://github.com/long-war-2/lwotc/pull/1728

**Comment by Iridar** [comment]
> it also cures those statuses that a Medikit would

It doesn't, though? What are you talking about?

**Comment by copyrite** [comment]
Some excerpts from `X2Ability_SpecialistAbilitySet.uc`:
```
static function X2AbilityTemplate RevivalProtocol()
{
	...
	Template.AddTargetEffect(RemoveAdditionalEffectsForRevivalProtocolAndRestorativeMist())
	...
}
```
```
static function X2Effect_RemoveEffects RemoveAdditionalEffectsForRevivalProtocolAndRestorativeMist()
{
	...

	foreach class'X2Ability_DefaultAbilitySet'.default.MedikitHealEffectTypes(HealType)
	{
		RemoveEffects.DamageTypesToRemove.AddItem(HealType);
	}

	return RemoveEffects;
}
```
 
And here's a demo with a Skirm and a Specialist. First, here's a burning Skirm: 
![20241229103021_1](https://github.com/user-attachments/assets/cf26854e-3fe4-4715-927a-2b83d51e114a)
We can see that Revival Protocol is unavailable:
![20241229103031_1](https://github.com/user-attachments/assets/d99e703f-7864-424a-919b-4f5120e53f07)
Now that the Skirm has been rendered unconscious by the Stun Lancer, Revival Protocol is available:
![20241229103112_1](https://github.com/user-attachments/assets/35f067bf-610c-433c-a9ac-0e88a4c9be12)
Finally, using Revival Protocol clears both the Unconscious and the Burning:
![20241229103126_1](https://github.com/user-attachments/assets/15875374-f9b6-4bc9-842e-c9ab1a4340e3)

**Comment by BlackDog86** [comment]
Yeah so this one is a bit tricky and 'what firaxis intended' is not altogether that transparent from the various bits of code & localisation - for reference:
Revival protocol:
![image](https://github.com/user-attachments/assets/5cfec6de-6295-406a-b7a3-7d48502a09d0)
Restoration:
![image](https://github.com/user-attachments/assets/29ac6406-5a5e-4f7b-925d-4be86749eaf9)

For me, I don't think these abilities were intended to do the same thing - the localization is quite explicit that revival protocol is meant to remove mental status effects and even lists (most of the stuff) it's supposed to remove, whereas Restoration just says 'heals and revives everyone as needed'.

In my view, the nicest way of dealing with it would be to remove this bit from RemoveAdditionalEffectsForRevivalProtocolAndRestorativeMist:
```
	foreach class'X2Ability_DefaultAbilitySet'.default.MedikitHealEffectTypes(HealType)
	{
		RemoveEffects.DamageTypesToRemove.AddItem(HealType);
	}
```

Then add:
```Template.AddMultiTargetEffect(RemoveAllEffectsByDamageType)``` 
on the restoration template. 

That would make it so that revival protocol can't target or heal medikit status effects / DOTs but restoration will basically cure anything (once the 'can't use on units with no missing HP' bug in the other issue is fixed at least).

**Comment by Iridar** [comment]
> Template.AddTargetEffect(RemoveAdditionalEffectsForRevivalProtocolAndRestorativeMist())

Dunno how I missed that 🤦
